packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts

```diff
@@ -342,3 +509,32 @@ function combineAndDedupeContainedResources(
 
   return dedupeContainedResources(combined);
 }
+
+async function getCwIdentifierSystems(patient: Patient): Promise<Set<string>> {
+  const existingPatient = await getCwPatientDataOrFail({
+    id: patient.id,
+    cxId: patient.cxId,
+  });
+
+  const cwLinks = existingPatient.data.links ?? [];
+  const cwIdentifierSystems = new Set<string>();
+  cwLinks.forEach(link => {
+    link.patient?.identifier?.forEach(identifier => {
+      if (identifier.system) {
+        const system = stripUrnPrefix(identifier.system);
+        cwIdentifierSystems.add(system);
+      }
+    });
+  });
+
+  return cwIdentifierSystems;
+}
+
+function isDocRefDuplicatedInCw(
+  docRef: DocumentReferenceWithMetriportId,
+  cwIdentifierSystems: Set<string>
+): boolean {
+  const homeCommunityId = docRef.homeCommunityId;
+  const repositoryUniqueId = docRef.repositoryUniqueId;
+  return cwIdentifierSystems.has(homeCommunityId) || cwIdentifierSystems.has(repositoryUniqueId);
```

Comment:
not following this logic entirely. The terminology is kinda confusing. Is this saying that if two docs (one from CQ and one from CW) came from the same facility, they are duplicates based on just ONE of `homeCommunityId` (which i think is external gw identifier) or `repositoryUniqueId` (which idr what this is)?